src/tests/networking_tests.rs

```diff
@@ -512,9 +512,16 @@ pub mod net_tests {
                         }
                     } else { //If the socket is in established conn., then we recv the data. If there's no data, then close the client socket.
                         let mut buf = sizecbuf(4);
-                        let recres = cage.recv_syscall(sockfd, buf.as_mut_ptr(), 4, 0);
-                        assert_eq!(recres & !4, 0); //This must be 0 or 4 to be correct, either the socket is good for recieving or it's closed
-                        if recres == 4 {
+                        let mut result :i32;
+                        loop {
+                            result = cage.recv_syscall(sockfd, buf.as_mut_ptr(), 4, 0);
+                            if result == -libc::EINTR {
+                                continue; // if the error was EINTR, retry the syscall
+                            }
+                            assert_eq!(result & !4, 0); //This must be 0 or 4 to be correct, either the socket is good for recieving or it's closed
+                            break;
+                        }
```

Comment:
```suggestion
                        loop {
                            result = cage.recv_syscall(sockfd, buf.as_mut_ptr(), 4, 0);
                            if result != -libc::EINTR {
                                break; // if the error was EINTR, retry the syscall
                            }
                            assert_eq!(result & !4, 0); //This must be 0 or 4 to be correct, either the socket is good for recieving or it's closed
                        }
```

Reply:
Yeah I agree with @kuzeyardabulut I had made similar suggestions in the last PR.

Reply:
Makes sense! But I think the suggested edit should also reorder the assertion to right before the break